scri/bms_transformations.py

```diff
@@ -0,0 +1,829 @@
+        boost_matrix = np.linalg.multi_dot([u, np.diag(s), u.conj().T])
+    else:
+        rotation_matrix = np.linalg.multi_dot([u, vh])
+        boost_matrix = np.linalg.multi_dot([vh.conj().T, np.diag(s), vh])
+    rotation = pure_spin_matrix_to_Lorentz(rotation_matrix, is_rotation=True)
+    boost = pure_spin_matrix_to_Lorentz(boost_matrix, is_rotation=False)
+    return Lorentz_transformation(rotation=rotation, boost=boost, order=output_order)
+
+def compute_conformal_factor(Lorentz, n_theta, n_phi):
+    """Compute the conformal factor.
+
+    Note that this is the inverse of Eq. (4.6) of 10.1063/1.1705135.
+    """
+    thetaprm_phiprm = sf.theta_phi(n_theta, n_phi)
+
+    [[a, b], [c, d]] = Lorentz_to_spin_matrix(Lorentz)
+
+    conformal = np.zeros((n_theta, n_phi), dtype=complex)
+    for j in range(n_theta):
+        for k in range(n_phi):
+            thetaprm_j, phiprm_k = thetaprm_phiprm[j, k]
+            if thetaprm_j == 0:
+                z = np.inf
+                conformal[j, k] = 1/(np.abs(a)**2+np.abs(c)**2)
+            else:
+                z = np.exp(1j*phiprm_k)/np.tan(thetaprm_j/2)
+                conformal[j, k] = (1+np.abs(z)**2)/(np.abs(a*z+b)**2+np.abs(c*z+d)**2)
+            
+    return conformal
+
+def nCr(n, r):
+    """n choose r.
+    """
+    if n >= 0 and r >= 0 and n - r >= 0:
+        return np.math.factorial(n)/(np.math.factorial(r)*np.math.factorial(n-r))
+    else:
+        return 0
+
+def A_matrix(L, M1, M2, N1, N2, spin_matrix):
+    """Compute the matrix that transforms Eq. (4.11) of 10.1063/1.1705135
+    to a Lorentz transformed frame.
+    """
+    A_value = 0
+    [[a, b], [c, d]] = spin_matrix
+    for i1 in range(L - M1 + 1):
+        for i2 in range(M1 + 1):
+            if not (i1 + i2) == N1:
+                continue
+            for i3 in range(L - M2 + 1):
+                for i4 in range(M2 + 1):
+                    if not (i3 + i4) == N2:
+                        continue
+                    A_value += nCr(L-M1,i1)*nCr(M1,i2)*nCr(L-M2,i3)*nCr(M2,i4)\
+                        *a**(L-M1-i1)*np.conjugate(a)**(L-M2-i3)*b**(i1)*np.conjugate(b)**(i3)\
+                        *c**(M1-i2)*np.conjugate(c)**(M2-i4)*d**(i2)*np.conjugate(d)**(i4)
+    
+    return A_value
+    
+def B_matrix(Lmax, L, M, M1, M2):
+    """Compute Eq. (4.14) of 10.1063/1.1705135.
+    """
+    if M1 + M != M2:
+        return 0
+    else:
+        ALM = (-1)**(L)*np.sqrt(np.math.factorial(L+M)*np.math.factorial(L-M)*(2*L+1)/(4*np.pi))
+        sum_contribution = 0
+        for p in range(min(Lmax-M1, L, L + M) + 1):
+            sum_contribution += (-1)**(p-M)*nCr(L,p)*nCr(L,p-M)*nCr(Lmax-L,Lmax-M1-p)
+        return ALM/np.sqrt(float(np.math.factorial(L)*np.math.factorial(L)))*sum_contribution
+    
+def transform_Y_LM(Y_LM, Lorentz, ell_max=None):
+    """Apply a Lorentz transformation to a spherical harmonic.
+
+    Parameters
+    ----------
+    f_LM: ndarray, dtype=complex
```

Comment:
Docs parameter name doesn't match what's above. I think this is actually an array-like with two (real) elements, meaning (L,M) -- is that right?